webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -26,26 +26,30 @@ export const getResults = ({
     id,
   });
 
-  const path = getAPIPath({ trigger, searchQuery, url });
   return createAPIRequest({
     controller,
-    path,
     searchQuery,
     trigger,
     id,
     dispatch,
+    url,
   });
 };
 
 let createAPIRequest = ({
   controller,
-  path,
   searchQuery,
   trigger,
   id,
   dispatch,
-}) =>
-  API.get(path)
+  url,
+}) => {
+  if (!url) {
+    requestFailure({ error: null, id, dispatch });
+    throw new Error('No API path was provided.');
```

Comment:
Why do you need to `throw` this error? In other words, who catch and respond to this error?
Shouldn't `return requestFailure({ error: null, id, dispatch });` is good enough?

Reply:
these errors that I `throw` will be visible only in the console with no need to catch or respond to.
the `requestFailure` is being called to indicate that there was a failure,
but the error is not always interesting for the user - as discussed here: https://github.com/theforeman/foreman/pull/6142#discussion_r242213533.

in `requestFailure` I am returning `{ error: null, ...}` so the error won't be visible.

Reply:
That comment was specifically regarding this case, but in the case of network errors i think it actually makes sense to tell the user it failed, rather than the console showing "unhandled promise rejection error".
Also, I think perhaps it makes sense to use `console.warn` in this case instead of throwing errors that aren't caught?

Reply:
IMO we should never throw errors if we have no intention to catch them as a golden rule, errors should be handled by us and not by the browser.

In redux actions, it doesn't make any sense to throw errors because we update the UI based on state changes and not based on event calls.

In other words:
```js
const MyCounter = ({ count, updateCount }) => {
  const addOne = () => {
    try {
      updateCount(count + 1);
    } catch (error) {
      // There is no much i can do here because I can't render the error from here
      // Instead throwing an error, it would make more sense to
      // update the state so it will cause the UI to get rerendered with error
    }
  };

  return (
    <div>
      <span>{count}</span>
      <button onClick={addOne}>Add one</button>
    </div>
  );
};
```

I agree with @tbrisker, `console.warn` would make more sense here.

Reply:
thanks guys, makes sense..update with your suggestion.. although I prefer to use `console.error` instead of a warning.. 
@tbrisker you think we should display the network error message?
@sharvit, I am going to update the action file structure to what you suggested.